### PR TITLE
[17.01] History Fixes

### PR DESCRIPTION
**Pull request by dannon**
xref #3337 
xref #3481

The fix for the toplevel 'hide|delete all hidden datasets' is a temporary reversion to the way we used to do this prior to the new history models.  It'll be replaced with new functionality in the history panel with the next release that supports more complex state selectors (all | hidden | deleted | none), and a batch mode dataset attribute toggle for these fields.

### CODE REVIEW

File: client/galaxy/scripts/mvc/history/history-view.js [hunk 1]
```diff
@@ -241,6 +241,7 @@ var HistoryView = _super.extend(
 
     /** override to render pagination also */
     renderItems: function( $whereTo ){
+        console.log("RENDER ITEMS");
```
Comment (martenson) [inline client/galaxy/scripts/mvc/history/history-view.js]: Do we need these console entries? Additionally there already was a commented out log so now we have 2.

File: tools/filters/random_lines_two_pass.py [hunk 2]
```diff
@@ -29,6 +29,8 @@ def get_random( line_offsets, num_lines ):
 
 
 def __main__():
+    import time
+    time.sleep(500000)
```
Comment (nsoranzo) [inline tools/filters/random_lines_two_pass.py]: These also should be removed.

Comment (dannon) [reply tools/filters/random_lines_two_pass.py]: Whoops.  That was debug code, good catch.

File: client/galaxy/scripts/mvc/history/history-view.js (hunk 1 above)
Comment (dannon) [reply client/galaxy/scripts/mvc/history/history-view.js]: No, definitely not, removing the debug garbage, one sec.

**Comment by dannon** [comment]
(pushing another change in a sec -- I inadvertently merged my stashed debug garbage here :( )

File: lib/galaxy/webapps/galaxy/controllers/history.py [hunk 3]
```diff
@@ -827,6 +827,22 @@ def share( self, trans, id=None, email="", **kwd ):
...
+        release after 17.01 """
+        action = kwd.get('user_action', None)
+        if action == 'delete':
+            for hda in trans.history.datasets:
+                if not hda.visible:
+                    hda.mark_deleted()
+            print 'deleting hidden'
+        elif action == 'unhide':
+            trans.history.unhide_datasets()
+            print 'unhiding hidden'
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/controllers/history.py]: Also this print and the one above can probably be removed.

Comment (dannon) [reply lib/galaxy/webapps/galaxy/controllers/history.py]: Thanks, yes, this was more of the debugging stash mentioned before :/

**Review by nsoranzo — APPROVED** [review]

**Comment by jmchilton** [comment]
Awesome - thanks for looking into that @dannon!

**Pull request merged**